internal/services/machinelearning/machine_learning_workspace_resource.go

```diff
@@ -315,6 +336,22 @@ func resourceMachineLearningWorkspaceCreateOrUpdate(d *pluginsdk.ResourceData, m
 		},
 	}
 
+	serverlessCompute := expandMachineLearningWorkspaceServerlessCompute(d.Get("serverless_compute").([]interface{}))
+	if serverlessCompute != nil {
+		if *serverlessCompute.ServerlessComputeNoPublicIP && serverlessCompute.ServerlessComputeCustomSubnet == nil && !networkAccessBehindVnetEnabled {
+			return fmt.Errorf(" Not supported to set `public_ip_enabled` to `false` without `subnet_id` when `public_network_access_enabled` is `false`")
+		}
+
+		if serverlessCompute.ServerlessComputeCustomSubnet == nil {
+			oldVal, newVal := d.GetChange("serverless_compute.0.public_ip_enabled")
+			if oldVal.(bool) && !newVal.(bool) {
+				return fmt.Errorf(" Not supported to update `public_ip_enabled` from `true` to `false` when `subnet_id` is null or empty")
+			}
+		}
```

Comment:
This logic looks like it belongs in a `CustomizeDiff`

Reply:
will add a `CustomizeDiff` for this

Reply:
Tried several ways to update this property, customers can still update it but it takes several steps. We can omit `CustomizeDiff``, cx can update the settings according to the error message.

1. Set the `serverless_compute.subnet_id` 
2. They may need to add the a private link for this machine learning workspace.
3. Update the `serverless_compute.public_ip_enabled`  from `true` to `false`


Error message cx may get:
```log
Error: creating/updating Workspace (Subscription: "<sub id>"
Resource Group Name: "<rg name>"
Workspace Name: "xz3workspace16"): unexpected status 400 (400 Bad Request) with error: ValidationError: Found errors while validating Subnet Id <subnet id>. Subnet Diagnostic Result = WorkspaceVnet:ErrorCode=WorkspaceVnetConflictErrorMessage=The workspace does not have public access and does not have a VNET. Pleas
e either add a VNET to the workspace or enable public access. For more information, please refer to https://docs.microsoft.com/en-us/azure/machine-learning/how-to-configure-private-link?tabs=python#add-a-private-endpoint-to-a-workspace.

```

Reply:
Thanks for the explanation @xuzhang3.

Is this something we can do in the resource by calling the `CreateOrUpdate` method twice in the same apply if `public_ip_enabled` changes from `true` -> `false`? 

e.g. First `CreateOrUpdate` call sets `subnet_id`
Second `CreateOrUpdate` call updated `public_ip_enabled` from `true` -> `false`

Reply:
no, the operation needs to be split into several steps

Reply:
Is that because of 
> 2. They may need to add the a private link for this machine learning workspace.

Is this an external thing that they need to configure? What is meant by _may_ are there situations where they don't need to?

Reply:
For example, I create a new MLW with following config and I want to set the `serverless_compute.public_ip_enabled` to `false`,  I need to update the `public_network_access_enabled` and `serverless_compute.subnet_id` and I need add a private endpoint for this MLW then I can update the  `serverless_compute.public_ip_enabled`, this is one of the scenario. Other scenario depends on the MLW configurations.

```hcl
resource "azurerm_machine_learning_workspace" "example" {
  name                    = "mymlwexample"
  location                = azurerm_resource_group.example.location
  resource_group_name     = azurerm_resource_group.example.name
  application_insights_id = azurerm_application_insights.example.id
  key_vault_id            = azurerm_key_vault.example.id
  storage_account_id      = azurerm_storage_account.example.id

#  public_network_access_enabled = true

  image_build_compute_name = "terraformComputeUpdate11"

  serverless_compute {
#    subnet_id         = azurerm_subnet.test.id
    public_ip_enabled = true
  }

  identity {
    type = "SystemAssigned"
  }
}
```